### PR TITLE
[Mellanox] Add support for platform SN5400

### PR DESCRIPTION
**Pull request by DavidZagury**
#### Why I did it
Add support for new platform x86_64-nvidia_sn5400-r0

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Added all config files that are needed to support SN5400.

#### How to verify it
Run Platform tests

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: device/mellanox/x86_64-nvidia_sn5400-r0/ACS-SN5400/buffers_defaults_t0.j2 [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{#
```
Comment (stephenxs) [inline device/mellanox/x86_64-nvidia_sn5400-r0/ACS-SN5400/buffers_defaults_t0.j2:1]: We can just make this file a symbol link to `x86_64-nvidia_sn5600-r0/ACS-SN5600/buffers_defaults_t0.j2`

File: device/mellanox/x86_64-nvidia_sn5400-r0/ACS-SN5400/pg_profile_lookup.ini [hunk 2]
```diff
@@ -0,0 +1,42 @@
+##
```
Comment (stephenxs) [inline device/mellanox/x86_64-nvidia_sn5400-r0/ACS-SN5400/pg_profile_lookup.ini:1]: this one can be changed to symbol link

**Review by stephenxs — APPROVED** [review]

File: device/mellanox/x86_64-nvidia_sn5400-r0/sensors.conf [hunk 3]
```diff
@@ -0,0 +1,368 @@
+##################################################################################
+# Copyright (c) 2019 - 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (liat-grozovik) [inline device/mellanox/x86_64-nvidia_sn5400-r0/sensors.conf:2]: @DavidZagury this is new file. header cannot be old.

**Review by liat-grozovik — CHANGES_REQUESTED** [review]

**Review by liat-grozovik — APPROVED** [review]

**Comment by dprital** [comment]
@yxieca , Can you please merge to 202311 ?

**Comment by yxieca** [comment]
> @yxieca , Can you please merge to 202311 ?

@dprital this change is not a bug fix. Can it wait for 202405 branch?

**Comment by dprital** [comment]
> > @yxieca , Can you please merge to 202311 ?
> 
> @dprital this change is not a bug fix. Can it wait for 202405 branch?

@liat-grozovik to comment on that.

**Pull request merged**